### DISCUSSION TITLE
How to load an HTML file

### DISCUSSION THREAD

**Discussion by hditano** [zauberzeug/nicegui#686] [Q&A] [answered]
Hello there:

I tried so far the following, but i'm unable to load the html file.

```py
ui.html('<div w3-include-html="my_map4.html"></div>')
ui.html('<link href="my_map4.html" rel="import" />')
```

and

```py
ui.html('<iframe src="/my_map4.html" title="mapa"></iframe>')
```

Unless there is something else that I can do?

Thx!!

**Comment by falkoschindler** [comment, marked as answer]
I would try something like this:

```py
with open('my_map.html') as f:
    code = f.read()
    ui.html(code)
```

**Reply by hditano** [reply]
> I would try something like this:
> 
> ```python
> with open('my_map.html') as f:
>     code = f.read()
>     ui.html(code)
> ```

Thx for your answer!!
I thought of that.
Here's the issue, I need to reload that file every 10 seconds or so.

Obviously i tried to take the route with ui.timer

```
ui.timer(interval=10.0, callback=reload_map()')
```

Is there any way to do it?

thx!!

**Reply by falkoschindler** [reply]
To call `reload_map` every 10 seconds, define the timer like this:
```py
ui.timer(interval=10.0, callback=reload_map)
```
See https://github.com/zauberzeug/nicegui/issues/648#issuecomment-1486549440 for an explanation of different ways to pass a function as an event handler.

**Comment by rodja** [comment]
The approaches

```html
<div w3-include-html="my_map4.html"></div>
<link href="my_map4.html" rel="import" />
```

are both not supported by modern browsers out of the box. The `iframe` approach works fine als long as you have defined a route to the html via FastAPI or [`ui.add_static_files`](https://nicegui.io/documentation#add_static_files_).
But Falkos suggestion above is still much better :-)

**Comment by hditano** [comment]
Thx to both of you for the answer.

I'm using Folium to update a static map, but i'm unable to make it update in NiceGui

This is what I tried to do, following the documentation:

1.  Pretty Much inside OMap it will create the static map, take the longitude/latitude values, it will create markers on each update and it will be saved as .html file.
But I can't make ui.add_body_html update on NiceGui with an ui.timer.

![image](https://user-images.githubusercontent.com/6069906/229220599-29a9e1a0-cd07-4d8f-ad8d-5ea95bda1b6d.png)

```
def main():
        .....
        with ui.row():
            ui.add_body_html(oMap.updateMap())
```


`
ui.timer(interval=2, callback=oMap.updateMap())
`

**Reply by falkoschindler** [reply]
You're still passing the function incorrectly. Should be `ui.timer(interval=2, callback=oMap.updateMap)`.
And better use `ui.html` instead of `ui.add_body_html` (which is not even a real UI element).

**Reply by falkoschindler** [reply]
Have you seen https://github.com/zauberzeug/nicegui/tree/main/examples/map? It doesn't use Folium, but might serve your needs nevertheless.
By the way, I guess NiceGUI needs a `ui.map` element...

**Reply by hditano** [reply]
> You're still passing the function incorrectly. Should be `ui.timer(interval=2, callback=oMap.updateMap)`. And better use `ui.html` instead of `ui.add_body_html` (which is not even a real UI element).

Using callback=oMap.updateMap 

will cause the following:

![image](https://user-images.githubusercontent.com/6069906/229221987-df20d65b-68c2-47c7-b1a2-357574903e54.png)


Not using add_body_html will cause the following:

![image](https://user-images.githubusercontent.com/6069906/229222329-9d5fc097-b5fa-4031-86f5-42aede1df8aa.png)

**Reply by hditano** [reply]
> Have you seen https://github.com/zauberzeug/nicegui/tree/main/examples/map? It doesn't use Folium, but might serve your needs nevertheless. By the way, I guess NiceGUI needs a `ui.map` element...

I'll take a look at it. I didn't really want to use any JS for something simple like this

**Comment by hditano** [comment]
I'll keep looking and find a solution to this.
